kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sequences/matchers.kt

```diff
@@ -49,12 +49,19 @@ fun <T> Sequence<T>.shouldHaveElementAt(index: Int, element: T) = this should ha
 fun <T> Sequence<T>.shouldNotHaveElementAt(index: Int, element: T) = this shouldNot haveElementAt(index, element)
 
 fun <T, S : Sequence<T>> haveElementAt(index: Int, element: T) = object : Matcher<S> {
-   override fun test(value: S) =
-      MatcherResult(
-         value.elementAt(index) == element,
-         { "Sequence should contain $element at index $index" },
-         { "Sequence should not contain $element at index $index" }
+   override fun test(value: S): MatcherResult {
+      val elementAtIndex = value.take(index + 1).mapIndexed { index, t -> index to t }.lastOrNull()
+      val passed = elementAtIndex?.let { it.first == index && it.second == element } ?: false
+      val description = elementAtIndex?.let {
+         if(it.first == index) ", but the value was different: ${it.second.print().value}."
```

Comment:
🟡 I think this pair being passed around should at least be desconstructed with proper names.

I want to suggest an alternative for this description with so many branches and magiic numbers, but it became hard to understand